tests/phpunit/api/v3/ContributionTest.php

```diff
@@ -1157,8 +1151,8 @@ public function testCreateUpdateContribution() {
 
     $this->assertEquals($contribution['values'][$contributionID]['contact_id'], $this->_individualId);
     $this->assertEquals($contribution['values'][$contributionID]['total_amount'], 110.00);
-    $this->assertEquals($contribution['values'][$contributionID]['financial_type_id'], $this->_financialTypeId);
-    $this->assertEquals($contribution['values'][$contributionID]['instrument_id'], $old_payment_instrument);
+    $this->assertEquals($contribution['values'][$contributionID]['financial_type_id'],$this->_financialTypeId );
+    $this->assertEquals($contribution['values'][$contributionID]['payment_instrument_id'], $old_payment_instrument);
```

Comment:
Unfortunately this breaks our 'test contract' - ie. that we will continue to support tested inputs & outputs in the api. In other cases we have added a function to the api to handled things like this e.g 

/**
 * Support for schema changes made in 4.2
 * The main purpose of the API is to provide integrators a level of stability not provided by
 * the core code or schema - this means we have to provide support for api calls (where possible)
 * across schema changes.
 */
function _civicrm_api3_event_create_legacy_support_42(&$params){
  if(!empty($params['payment_processor_id'])){
    $params['payment_processor'] = CRM_Core_DAO::VALUE_SEPARATOR . $params['payment_processor_id'] . CRM_Core_DAO::VALUE_SEPARATOR;
  }
}